src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/Helpers.cs

```diff
@@ -118,6 +118,32 @@ public static byte[] FixupKeyParity(this byte[] key)
             }
             return oddParityKey;
         }
+
+        public static void ConvertIntToByteArray(uint dwInput, ref byte[] counter)
```

Comment:
> so it saves 11 array allocations by just writing into the same one over and over.

Why does it need to be ref for that?  It can just be ```byte[] counter``` instead of ```ref byte[] counter```... the method never assigns to counter, only to counter[whatever].